src/main/java/duke/models/serializable/TaskSerializable.java

```diff
@@ -0,0 +1,111 @@
+package duke.models.serializable;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.regex.Pattern;
+
+import duke.exceptions.DukeException;
+import duke.models.task.Deadline;
+import duke.models.task.Event;
+import duke.models.task.Task;
+import duke.models.task.TaskType;
+import duke.models.task.ToDo;
+import duke.utils.DukeValidator;
+
+/**
+ * Encapsulates the logic for serializing and deserializing a {@link Task}.
+ *
+ * @author Emily Ong Hui Qi
+ */
+public class TaskSerializable extends Serializable<Task> {
+    private static final Pattern MATCH_TASK_DATA = Pattern.compile("(?<taskType>[TDE])\\s\\|\\s"
+            + "(?<taskCompleted>[01])\\s\\|\\s"
+            + "(?<taskDescription>(.+))(\\s\\|\\s"
+            + "(?<taskMeta>(.+)))?"
+    );
+
+    private static final String ERROR_UNKNOWN_TASK_TYPE = "Unknown task type %s!";
+
+    private final TaskType taskType;
```

Comment:
Appreciate the use of camel style for variables throughout the project.

Reply:
Thanks!